arch/arm64/core/cache.c

```diff
@@ -75,10 +75,29 @@ int arch_dcache_range(void *addr, size_t size, int op)
 
 	line_size = arch_dcache_line_size_get();
 
+	/*
+	 * For the data cache invalidate operation, clean and invalidate
+	 * the partial cache lines at both ends of the given range to
+	 * prevent data corruption.
+	 */
```

Comment:
Can you make a numerical example in terms of addresses showing the problem with the current implementation?

Reply:
For example:
There are 4 consecutive 16-byte buffers, which can be cached in a line (let's assume the cache line size is 64 bytes) like below.
+----------------+--------------+--------------+----------------+
| buffer 0 (dirty) |     buffer 1     |     buffer 2     | buffer 3 (dirty) |
+----------------+--------------+--------------+----------------+
when invalidate the buffer 1 and 2, the full cache line will be invalidated, if the buffer 0 and buffer 3 is dirty, then the data in them will be lost.

Reply:
Ok, but why the `buffer 0` and `buffer 3` were not flushed already? This looks to me a very custom fix to a very special situation.

Reply:
IMHO, if we use `K_CACHE_INVD`, we need to be aware that the dirty cache will be dropped. If we want the dirty buffer to be written back, should we use `K_CACHE_WB_INVD` instead?

Reply:
> Ok, but why the `buffer 0` and `buffer 3` were not flushed already? This looks to me a very custom fix to a very special situation.

I think this is better centrally handled in cache maintain functions than in the various applications, if a application developer doesn't aware of this issue, it will cause a very hard debugging.  Furthermore, for portability of the application, it has to aware of and deal with different cache line size, if handle this in applications, such as, with 64-Byte cache line before invalidating `buffer 1` both the dirty `buffer 0` and `buffer 3` have to be cleaned, while with 32-Byte cache line, only `buffer 0` need to be cleaned. Obviously it will increase the cost of developing and maintenance.

Reply:
> IMHO, if we use `K_CACHE_INVD`, we need to be aware that the dirty cache will be dropped. If we want the dirty buffer to be written back, should we use `K_CACHE_WB_INVD` instead?

This is also a simple solution, but actually only the non-aligned first/last cache line need to be cleaned and other lines 
still keep invalidation, so I chose this way.